File: blockchain_integration/pi_network/pi-trade-nexus/smart-contract-based-trade-finance/scripts/deploy.js
const HDWalletProvider = require("truffle-hdwallet-provider");
const Web3 = require("web3");
const { abi, bytecode } = require("../contracts/TradeFinance.json");
const { abi: tokenAbi, bytecode: tokenBytecode } = require("../contracts/PiTradeToken.json");

const provider = new HDWalletProvider(
  "0x1234567890abcdef", // Your Ethereum wallet private key
  "https://mainnet.infura.io/v3/YOUR_PROJECT_ID" // Your Infura project ID
);

const web3 = new Web3(provider);

async function deploy() {
  const accounts = await web3.eth.getAccounts();
  const deployer = accounts[0];

  console.log("Deploying TradeFinance contract...");
  const tradeFinanceContract = new web3.eth.Contract(abi);
  const tradeFinanceTx = tradeFinanceContract.deploy({ data: bytecode });
  const tradeFinanceReceipt = await tradeFinanceTx.send({ from: deployer, gas: 5000000 });
  const tradeFinanceAddress = tradeFinanceReceipt.contractAddress;
  console.log(`TradeFinance contract deployed to ${tradeFinanceAddress}`);

  console.log("Deploying PiTradeToken contract...");
  const piTradeTokenContract = new web3.eth.Contract(tokenAbi);
  const piTradeTokenTx = piTradeTokenContract.deploy({ data: tokenBytecode });
  const piTradeTokenReceipt = await piTradeTokenTx.send({ from: deployer, gas: 5000000 });
  const piTradeTokenAddress = piTradeTokenReceipt.contractAddress;
  console.log(`PiTradeToken contract deployed to ${piTradeTokenAddress}`);

  console.log("Setting PiTradeToken address in TradeFinance contract...");
  const tradeFinanceInstance = new web3.eth.Contract(abi, tradeFinanceAddress);
  const setPiTradeTokenTx = tradeFinanceInstance.methods.setPiTradeToken(piTradeTokenAddress).send({ from: deployer, gas: 2000000 });
  await setPiTradeTokenTx;
  console.log("PiTradeToken address set in TradeFinance contract");
}

deploy();

File: blockchain_integration/pi_network/pi-trade-nexus/tools/deploy.py
import os
import json
from web3 import Web3
from web3.auto import w3
from eth_account import Account

# Load configuration from environment variables
NETWORK = os.environ['NETWORK']
PRIVATE_KEY = os.environ['PRIVATE_KEY']
CONTRACT_ADDRESS = os.environ['CONTRACT_ADDRESS']

# Set up Web3 provider
w3 = Web3(Web3.HTTPProvider(f'https://{NETWORK}.infura.io/v3/{os.environ["INFURA_PROJECT_ID"]}'))

# Load contract ABI and bytecode
with open('contracts/PiTradeToken.sol/PiTradeToken.abi', 'r') as f:
    abi = json.load(f)
with open('contracts/PiTradeToken.sol/PiTradeToken.bin', 'r') as f:
    bytecode = f.read()

# Deploy contract
account = Account.from_key(PRIVATE_KEY)
tx_hash = w3.eth.contract(abi=abi, bytecode=bytecode).constructor().transact({'from': account.address, 'gas': 2000000})
tx_receipt = w3.eth.wait_for_transaction_receipt(tx_hash)

# Print contract address
print(f'Contract deployed to {tx_receipt.contractAddress}')
